pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -159,6 +145,29 @@ func (v *fleetResourceValidator) handleNamespace(req admission.Request) admissio
 	return admission.Allowed("namespace name doesn't begin with fleet/kube prefix so we allow all operations on these namespaces")
 }
 
+// handleResource allows/denies request to modify resource after validation.
+func (v *fleetResourceValidator) handleResource(req admission.Request) admission.Response {
+	return validation.ValidateUserForResource(req.Kind.Kind, types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, v.whiteListedUsers, req.UserInfo)
+}
+
+// handleNamespaceResources allows/denies request to modify namespaced resources after validation.
+func (v *fleetResourceValidator) handleNamespacedResources(ctx context.Context, req admission.Request) admission.Response {
+	resourceMsgFormat := fmt.Sprintf("handling %s resource", req.Kind.Kind)
+	reqGroupVersion := schema.GroupVersion{Group: req.Resource.Group, Version: req.Resource.Version}
+	namespacedName := types.NamespacedName{Name: req.Name, Namespace: req.Namespace}
+	var response admission.Response
+	if reqGroupVersion == appsV1GV || reqGroupVersion == coreV1GV || reqGroupVersion == rbacV1GV || reqGroupVersion == batchV1GV {
+		klog.V(2).InfoS(resourceMsgFormat, "GVK", req.Kind, "namespacedName", namespacedName, "operation", req.Operation)
+		response = v.handleResource(req)
+	} else if reqGroupVersion == fleetV1Alpha1GV {
+		if req.Kind.Kind == "InternalMemberCluster" {
```

Comment:
Hi Arvind! Is there any particular reason why `InternalMemberCluster` is singled out here? We also have a number of other v1alpha1 resources that needs guarding I believe.